src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProject.cs

```diff
@@ -144,18 +144,17 @@ public async Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, stri
                 case "TargetPath": _project.OutputFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetRefPath": _project.OutputRefFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetFrameworkIdentifier": _targetFrameworkManager.UpdateIdentifierForProject(_project.Id, valueOrNull); break;
+                case "IntermediateAssembly": _project.CompilationOutputAssemblyFilePath = GetFullyQualifiedPath(valueOrNull); break;
             }
         }
 
-        // Workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1830960
-        _project.CompilationOutputAssemblyFilePath = _project.OutputFilePath;
-
         string? GetFullyQualifiedPath(string? propertyValue)
         {
-            Contract.ThrowIfNull(_project.FilePath, "We don't have a project path at this point.");
+            var projectPath = Path.GetDirectoryName(_project.FilePath);
+            Contract.ThrowIfNull(projectPath, "We don't have a project path at this point.");
 
             if (propertyValue is not null)
-                return Path.Combine(_project.FilePath, propertyValue);
+                return Path.Combine(projectPath, propertyValue);
```

Comment:
This was an existing bug, but I guess `IntermediateOutputPath` is the first property that actually has a relative path in it. When calling `Path.Combine("C:\blah", "C:\blah2")` it helpfully returns the latter parameter.

Reply:
Yikes, good find.